chiron/integrators.py

```diff
@@ -137,10 +144,29 @@ def run(
         b = jnp.sqrt(1 - jnp.exp(-2 * collision_rate_unitless * stepsize_unitless))
 
         # Initialize velocities
-        if self.velocities is None:
-            v0 = sigma_v * random.normal(key, x0.shape)
+        if self.reinitialize_velocities:
+            # v0 = sigma_v * random.normal(key, x0.shape)
+            from .utils import initialize_velocities
+
+            sampler_state.velocities = initialize_velocities(
+                temperature, potential.topology, key
+            )
+            self.initialize_velocities = False
+
+        elif self.initialize_velocities:
+            # v0 = sigma_v * random.normal(key, x0.shape)
+            from .utils import initialize_velocities
+
+            sampler_state.velocities = initialize_velocities(
+                temperature, potential.topology, key
+            )
+            self.initialize_velocities = False
         else:
```

Comment:
My thought was trying to be very explicit in what the code was doing (rather than having it implicitly initialize velocities if they were not defined).  I think either way is fine; more important is that we need to check to see if the shape of x0 matches the velocities in the sampler state in case we have a gcmc move.